### PR TITLE
fix: remove registered tokens when changing network settings

### PR DESCRIPTION
**Pull request by alexruzenhack**
### Acceptance Criteria
- It should remove registered tokens when a user changes the network settings
- It should closes https://github.com/HathorNetwork/internal-issues/issues/286

It depends on:
- https://github.com/HathorNetwork/hathor-wallet-lib/pull/648

### Security Checklist
- [x] Make sure you do not include new dependencies in the project unless strictly necessary and do not include dev-dependencies as production ones. More dependencies increase the possibility of one of them being hijacked and affecting us.

### CODE REVIEW

File: src/sagas/networkSettings.js [hunk 1]
```diff
@@ -289,6 +289,8 @@ export function* persistNetworkSettings(action) {
     return;
   }
 
+  // Remove registered tokens
+  STORE.removeItem(REGISTERED_TOKENS_KEY);
```
Comment (pedroferreira1) [inline src/sagas/networkSettings.js:293]: Does it remove from the wallet storage as well, in the `wallet.storage`?

Comment (alexruzenhack) [reply src/sagas/networkSettings.js:293]: This one: https://github.com/HathorNetwork/internal-issues/issues/285

Comment (alexruzenhack) [reply src/sagas/networkSettings.js:293]: Yes, it does.

Comment (alexruzenhack) [reply src/sagas/networkSettings.js]: I refactored it.

**Review by andreabadesso — APPROVED** [review]

File: src/sagas/networkSettings.js [hunk 2]
```diff
@@ -289,6 +289,8 @@ export function* persistNetworkSettings(action) {
     return;
   }
 
+  // Remove registered tokens only
+  yield call([wallet.storage, wallet.storage.cleanStorage], false, false, true)
   // Stop wallet and clean its storage without clean its access data.
   wallet.stop({ cleanStorage: true, cleanAddresses: true });
```
Comment (pedroferreira1) [inline src/sagas/networkSettings.js:295]: The method `stop` from the wallet already calls the `cleanStorage`. With this change we are calling it twice.

We should add a new parameter to the `stop` function to support `cleanTokens` as well.

Comment (alexruzenhack) [reply src/sagas/networkSettings.js:295]: Oh, ok! I will open an issue for it.

Comment (alexruzenhack) [reply src/sagas/networkSettings.js:295]: See https://github.com/HathorNetwork/internal-issues/issues/290

Comment (pedroferreira1) [reply src/sagas/networkSettings.js:295]: I think we should do this in this PR, it's a problem to merge like this

Comment (andreabadesso) [reply src/sagas/networkSettings.js:305]: So, are we going to do this in this PR? Was it solved in a different way?

**Review by andreabadesso — CHANGES_REQUESTED** [review]

File: src/sagas/networkSettings.js (hunk 2 above)
Comment (alexruzenhack) [reply src/sagas/networkSettings.js]: Done

**Review by andreabadesso — APPROVED** [review]

File: src/store.js [hunk 3]
```diff
@@ -143,6 +143,22 @@ class HybridStore extends MemoryStore {
...
+
+  /**
+   * Clean the storage.
+   * @param {boolean} cleanHistory if we should clean the transaction history.
+   * @param {boolean} cleanAddresses if we should clean the addresses.
+   * @param {boolean} cleanTokens if we should clean the registered tokens.
+   * @async
+   * @returns {Promise<void>}
+   */
+  async cleanStorage(cleanHistory = false, cleanAddresses = false, cleanTokens = false) {
```
Comment (pedroferreira1) [inline src/store.js:155]: Why do you need this method?

Comment (alexruzenhack) [reply src/store.js:155]: Oh, this is needed to also clean the app's AsyncStorage managed by the `HybridStore`.

**Review by pedroferreira1 — APPROVED** [review]

**Pull request merged**